Numpy 2.4 compatibility

It looks like ESMValCore is not compatible with numpy 2.4.

This change https://numpy.org/doc/stable/release/2.4.0-notes.html#raise-typeerror-on-attempt-to-convert-array-with-ndim-0-to-scalar, results in the following test failures:

```
FAILED tests/integration/preprocessor/_concatenate/test_concatenate.py::test_concatenation_with_aux_factory - TypeError: only 0-dimensional arrays can be converted to Python scalars
FAILED tests/integration/preprocessor/_concatenate/test_concatenate.py::test_relax_concatenation[4] - TypeError: only 0-dimensional arrays can be converted to Python scalars
FAILED tests/integration/preprocessor/_concatenate/test_concatenate.py::test_relax_concatenation[5] - TypeError: only 0-dimensional arrays can be converted to Python scalars
FAILED tests/integration/preprocessor/_concatenate/test_concatenate.py::TestConcatenate::test_concatenate - TypeError: only 0-dimensional arrays can be converted to Python scalars
FAILED tests/integration/preprocessor/_concatenate/test_concatenate.py::TestConcatenate::test_concatenate_by_experiment_first - TypeError: only 0-dimensional arrays can be converted to Python scalars
FAILED tests/integration/preprocessor/_concatenate/test_concatenate.py::TestConcatenate::test_concatenate_differing_attributes - TypeError: only 0-dimensional arrays can be converted to Python scalars
FAILED tests/integration/preprocessor/_concatenate/test_concatenate.py::TestConcatenate::test_concatenate_remove_unwanted_attributes - TypeError: only 0-dimensional arrays can be converted to Python scalars
FAILED tests/integration/preprocessor/_concatenate/test_concatenate.py::TestConcatenate::test_concatenate_remove_unwanted_attributes_from_coords - TypeError: only 0-dimensional arrays can be converted to Python scalars
FAILED tests/integration/preprocessor/_concatenate/test_concatenate.py::TestConcatenate::test_concatenate_with_iris_exception - TypeError: only 0-dimensional arrays can be converted to Python scalars
FAILED tests/integration/preprocessor/_concatenate/test_concatenate.py::TestConcatenate::test_concatenate_with_order - TypeError: only 0-dimensional arrays can be converted to Python scalars
FAILED tests/integration/preprocessor/_concatenate/test_concatenate.py::TestConcatenate::test_concatenate_with_overlap - TypeError: only 0-dimensional arrays can be converted to Python scalars
FAILED tests/integration/preprocessor/_concatenate/test_concatenate.py::TestConcatenate::test_concatenate_with_overlap_2 - TypeError: only 0-dimensional arrays can be converted to Python scalars
FAILED tests/integration/preprocessor/_concatenate/test_concatenate.py::TestConcatenate::test_concatenate_with_overlap_3 - TypeError: only 0-dimensional arrays can be converted to Python scalars
FAILED tests/integration/preprocessor/_concatenate/test_concatenate.py::TestConcatenate::test_concatenate_with_overlap_same_start - TypeError: only 0-dimensional arrays can be converted to Python scalars
FAILED tests/integration/preprocessor/_concatenate/test_concatenate.py::TestConcatenate::test_convert_calendar_concatenate_with_overlap - TypeError: only 0-dimensional arrays can be converted to Python scalars
FAILED tests/integration/preprocessor/_concatenate/test_concatenate.py::TestConcatenate::test_fail_metadata_differs - TypeError: only 0-dimensional arrays can be converted to Python scalars
```
This is a problem with the implementation of the `concatenate` preprocessor function.

I suspect #2934 is also new because of numpy 2.4.

--- a/esmvalcore/preprocessor/_concatenate.py
+++ b/esmvalcore/preprocessor/_concatenate.py
@@ -164,7 +164,9 @@ def _fix_calendars(cubes: Sequence[Cube]) -> None:
         [calendars.count(calendar) for calendar in unique_calendars],
     )
     calendar_index = int(
-        np.argwhere(calendar_ocurrences == calendar_ocurrences.max()),
+        np.argwhere(calendar_ocurrences == calendar_ocurrences.max()).ravel()[
+            0
+        ],
     )
 
     for cube in cubes:

--- a/tests/unit/preprocessor/test_shared.py
+++ b/tests/unit/preprocessor/test_shared.py
@@ -439,7 +439,7 @@ def test_try_adding_calculated_cell_area():
             da.ma.masked_array(np.arange(2), np.arange(2)),
         ),
         (
-            np.ma.masked_array(np.arange(2), mask=[1, 0]),
+            np.ma.masked_array(np.arange(2), mask=np.array([1, 0])),
             da.arange(2),
             (0,),
             da.ma.masked_array(np.arange(2), np.ones(2)),
@@ -456,7 +456,10 @@ def test_try_adding_calculated_cell_area():
             np.arange(2),
             np.ones((3, 2)),
             (1,),
-            np.ma.masked_array(np.ones((3, 2)), mask=[[0, 1], [0, 1], [0, 1]]),
+            np.ma.masked_array(
+                np.ones((3, 2)),
+                mask=np.array([[0, 1], [0, 1], [0, 1]]),
+            ),
         ),
     ],
 )

